veps/sig-storage/incremental-backup.md

```diff
@@ -367,6 +400,103 @@ Disadvantages:
 
+## Pull Mode M.O. and Alternatives
+
+Ingress to the virt-launcher Pod is currently impossible because all incoming traffic is redirected to the guest VM. To address this limitation, several alternative approaches have been considered.
+
+### Chosen M.O. - Backup Proxy Pod
+
+The backup process is initiated and the Unix socket is created. The virt-launcher then opens a bidirectional channel to the socket and continually attempts to connect to the backup proxy.  
+Meanwhile, the backup proxy starts and listens on a predetermined port exposed via a Kubernetes Service for the virt-launcher to connect to. Once the connection succeeds and the bidirectional channel is established, the backup proxy creates another bidirectional channel by mapping its connection to a second port, also exposed through the same Service.
+
+The `VirtualMachineBackupStatus` is then updated with the exposed endpoint and port that clients can use to interact with the backup stream.  
+Clients may optionally expose external access to the Service (e.g., by using a LoadBalancer Service type), or access it internally via the Service’s ClusterIP.
+
+This M.O was chosen because it preserves pull-mode semantics, requires no changes to the virt-launcher networking model, and introduces minimal API changes.
+
+In contrast:
+- Network Push M.O. - breaks several pull-mode features and requires extending the hotplug API.
+- Ingress - requires two new internal APIs (secret hotplug and conditional networking changes), significantly expanding the scope of this VEP.
+
+```mermaid
+graph TD
+    subgraph "Kubernetes Cluster"
+        subgraph "virt-launcher"
+            CH[Channel manager]
+            US[NBD Unix Socket]
+        end
+
+        subgraph "Backup Proxy Pod"
+            BP[Backup Proxy]
+        end
+
+        SVC[Kubernetes Service]
+    end
+
+    Client["NBD Client"]
+
+    %% 1: virt-launcher connects to NBD Unix socket
+    CH -- "1. Creates bidirectional channel" <--> US
+
+    %% 2: virt-launcher connects to backup proxy via Service
+    CH -- "2. Connects to internal port" --> SVC -- Internal Port <--> BP
+
+    %% 3: Client accesses using external port
+    Client -- "3. Client accesses using external port" --> SVC -- External Port <--> BP
+```
+
+**Security considerations and approaches**
+1. TLS with client authentication is used to secure the transport layer and verify client identity. The backup proxy exposes a TLS-enabled endpoint using its own certificate and key. The client provides a Secret containing the CA certificate and the expected Common Name (CN) of the client certificate. The client implementation remains minimal and does not perform server certificate verification, relying solely on TLS encryption and server-side client authentication to establish a trusted channel.
+2. Securing the backup proxy endpoint may also be delegated to the cluster operator, using an L4 ingress that provides traffic encryption and routes to the exposed Service.
```

Comment:
What about cross-namespace access to this endpoint? Will the backup controller create the necessary **NetworkPolicy** object for a backup application running in another namespace to connect to the NBD service? It may be necessary to provide the backup application's desired namespace in the input arguments in order to set the NamespaceSelector field of an ingress NetworkPolicyPeer.

Ideally the NetworkPolicy object should be owned by your associated runtime pod so that it (and the Service) can be cleaned up properly.